providers/universal-provider/src/providers/cardano.ts

```diff
@@ -0,0 +1,119 @@
+import HttpConnection from "@walletconnect/jsonrpc-http-connection";
+import { JsonRpcProvider } from "@walletconnect/jsonrpc-provider";
+import Client from "@walletconnect/sign-client";
+import { EngineTypes, SessionTypes } from "@walletconnect/types";
+import EventEmitter from "events";
+import {
+  IProvider,
+  RequestParams,
+  RpcProvidersMap,
+  SessionNamespace,
+  SubProviderOpts,
+} from "../types";
+import { getRpcUrl } from "../utils";
+
+class CardanoProvider implements IProvider {
+  public name = "cip34";
+  public client: Client;
+  public httpProviders: RpcProvidersMap;
+  public events: EventEmitter;
+  public namespace: SessionNamespace;
+  public chainId: string;
+
+  constructor(opts: SubProviderOpts) {
+    this.namespace = opts.namespace;
+    this.events = opts.events;
+    this.client = opts.client;
+    this.chainId = this.getDefaultChainId();
+    this.httpProviders = this.createHttpProviders();
+  }
+
+  public updateNamespace(namespace: SessionTypes.Namespace) {
+    this.namespace = Object.assign(this.namespace, namespace);
+  }
+
+  public requestAccounts(): string[] {
+    return this.getAccounts();
+  }
+
+  private getAccounts(): string[] {
+    const accounts = this.namespace.accounts;
+    if (!accounts) {
+      return [];
+    }
+
+    return (
+      accounts
+        // get the accounts from the active chain
+        .filter((account) => account.split(":")[1] === this.chainId.toString())
+        // remove namespace & chainId from the string
+        .map((account) => account.split(":")[2]) || []
+    );
+  }
+
+  private createHttpProviders(): RpcProvidersMap {
+    const http = {};
+    this.namespace.chains.forEach((chain) => {
```

Comment:
This needs to pass the rpcURL in directly from the rpcmap. GetChainID breaks on this.